t/lib/TestDB.pm

```diff
@@ -12,11 +12,37 @@ sub new {
 
     open my $fh, '<', 'share/test_data.sql';
 
+    my $continued_line = "";
+    my $last_line_was_continued = 0;
+
     while (<$fh>) {
         $ENV{DBIC_TRACE}
           and warn $_;
-
-        $schema->storage->dbh_do( sub { $_[1]->do($_) } );
+	my $line = $_;
+
+	# Remove comments from SQL-File
+	my $comment_start = index ($line, '--');
+	$line = substr ($line, 0, $comment_start);
+    # Remove trailing and leading whitespaces
+    $line =~ s/^\s+|\s+$//g;
+	next if $line =~ /^\s*$/;
```

Comment:
After removing trailing whitespace any line containing only whitespace now equals `"\n"`. You could `chomp()` all input lines at first and then `length($line) or next`.